### PR TITLE
feat: build UBI-based image

### PR DESCRIPTION
**Pull request by aboguszewski-sumo**
This is more like a PoC than ready PR:
- only linux/amd64 was added
- dev builds are only to confirm that it builds successfully: IMO we should only have this enabled for releases
- `LATEST_TAG_FIPS_SUFFIX` was used as a temporal solution: imo we should rename it just to `LATEST_TAG_SUFFIX` and use it as usual

### CODE REVIEW

**Comment by swiatekm-sumo** [comment]
> dev builds are only to confirm that it builds successfully: IMO we should only have this enabled for releases

I disagree, anything we build for releases should also be built for dev. Otherwise we're never going to catch breakage.

The approach in this PR looks reasonable to me in general.

**Comment by sumo-drosiek** [comment]
>     * `LATEST_TAG_FIPS_SUFFIX` was used as a temporal solution: imo we should rename it just to `LATEST_TAG_SUFFIX` and use it as usual

Can we rename it in this PR, or another one and merge it before this PR?

File: ci/build-push-multiplatform.sh [hunk 1]
```diff
@@ -82,7 +82,7 @@ function build_push() {
     esac
 
     local TAG
-    readonly TAG="${REPO_URL}:${BUILD_TAG}-${BUILD_ARCH}"
+    readonly TAG="${REPO_URL}:${BUILD_TAG}${LATEST_TAG_FIPS_SUFFIX}-${BUILD_ARCH}"
```
Comment (sumo-drosiek) [inline ci/build-push-multiplatform.sh:85]: Does it affect current builds (fips especially)?

**Comment by astencel-sumo** [comment]
> > dev builds are only to confirm that it builds successfully: IMO we should only have this enabled for releases
> 
> I disagree, anything we build for releases should also be built for dev. Otherwise we're never going to catch breakage.

I agree we should build, but we probably don't need to push the non-release UBI images to an image repository, right?

File: Dockerfile_ubi [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+RUN microdnf update && microdnf install systemd
+
+ENV TAG $BUILD_TAG
+ARG USER_UID=10001
+USER ${USER_UID}
+ENV HOME /etc/otel/
+
+COPY otelcol-sumo /
+COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
+COPY --from=directories --chown=${USER_UID}:${USER_UID} /etc/otel/ /etc/otel/
```
Comment (astencel-sumo) [inline Dockerfile_ubi:38]: Why is this directory copied from Alpine and not just created here?

Comment (aboguszewski-sumo) [reply Dockerfile_ubi:38]: see https://github.com/SumoLogic/sumologic-otel-collector/pull/1440#discussion_r1484357187

File: ci/build-push-multiplatform.sh (hunk 1 above)
Comment (sumo-drosiek) [reply ci/build-push-multiplatform.sh:85]: We can, but what was fips tag before change wand what is it now?

**Comment by swiatekm-sumo** [comment]
> > > dev builds are only to confirm that it builds successfully: IMO we should only have this enabled for releases
> > 
> > 
> > I disagree, anything we build for releases should also be built for dev. Otherwise we're never going to catch breakage.
> 
> I agree we should build, but we probably don't need to push the non-release UBI images to an image repository, right?

We currently do it for all builds on `main`, which I think is valid, as it makes it much easier to run external tests on dev versions.

File: ci/build-push-multiplatform.sh (hunk 1 above)
Comment (aboguszewski-sumo) [reply ci/build-push-multiplatform.sh:85]: Okay, it seems that it does not affect it, but I bugged it and this is why there are failures. I will take care of that.

File: Dockerfile_ubi [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+## Build RedHat compliant image
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.9
+
+ENV SUMMARY="UBI based sumologic-otel-collector" \
+    DESCRIPTION="Sumo Logic Distribution for OpenTelemetry Collector is a Sumo Logic-supported distribution of the OpenTelemetry Collector. It is a single agent to send logs, metrics and traces to Sumo Logic."
+
+LABEL name="sumologic-otel-collector" \
+      vendor="Sumo Logic" \
+      version="$BUILD_TAG" \
+      release="4" \
```
Comment (swiatekm-sumo) [inline Dockerfile_ubi:14]: What is this label for?

Comment (aboguszewski-sumo) [reply Dockerfile_ubi:14]: I wish I knew! This was already in the original Dockerfile for ubi, so I left it: https://github.com/SumoLogic/sumologic-openshift-images/blob/main/sumologic-otel-collector/0.57.2-sumo-1/Dockerfile

Comment (swiatekm-sumo) [reply Dockerfile_ubi:14]: @kkujawa-sumo do you know?

Comment (swiatekm-sumo) [reply Dockerfile_ubi:14]: Looking at this: https://developers.redhat.com/articles/2021/11/11/best-practices-building-images-pass-red-hat-container-certification#best_practice__6__embed_identifying_information_inside_your_image, this should probably just be `0`.

We should probably also apply security updates as per section 5 of that guide.

Comment (aboguszewski-sumo) [reply Dockerfile_ubi:14]: Doesn't `microdnf update` already apply these updates? The command in the section 5 does not work with `microdnf` (some unsupported flags).

Comment (aboguszewski-sumo) [reply Dockerfile_ubi:14]: Also, what should be equal to `0`? The `release` label?

**Review by swiatekm-sumo — APPROVED** [review]
LGTM, although we should add UBI FIPS images in a subsequent PR.

**Pull request merged**